Components/ApplePayDirect/Services/ApplePayButtonBuilder.php

```diff
@@ -90,8 +100,17 @@ public function addButtonStatus(Enlight_Controller_Request_Request $request, Enl
             if ($isRiskManagementBlocked) {
                 $isActive = false;
             }
-        }
 
+            # if a customer has esd products in the basket, check if
+            # the customer is logged in with a full customer account
+            $hasEsdProductsInBasket = $controller === 'checkout' && $this->basketHasEsdProducts();
```

Comment:
code looks working
but somehow i think we shouldnt keep it like this.
specific logic, inline strings...this seems to be at least a private function or maybe even moved to another class with unit tests

Reply:
As you have seen further on, there are unit tests. But sure, I can put this in a separate class.

Reply:
its okay like this thx